src/monster.h

```diff
@@ -437,7 +437,14 @@ class monster : public Creature
         character_id mounted_player_id; // id of player that is mounting this creature ( for save/load )
         character_id dragged_foe_id; // id of character being dragged by the monster
         cata::optional<item> tied_item; // item used to tie the monster
+        cata::optional<item> tack_item; // item representing saddle and reins and such
+        cata::optional<item> armor_item; // item of armor the monster may be wearing
+        cata::optional<item> storage_item; // storage item for monster carrying items
         cata::optional<item> battery_item; // item to power mechs
```

Comment:
This is a problem, this means *every* monster object will reserve enough space for these 5 items unconditionally.
To be clear, the two existing instances of cata::optional<item> are also a problem, but they slipped in without notice.
These can be std::unique_ptr<item> instead.

Reply:
How should I handle the `monster` copy constructor? Is deep copying for the `unique_ptr`s going to be a performance concern?

Reply:
Make it a `value_ptr` (it has a build in copy-constructor). Performance of copy is not a concern, it happens rarely enough.